src/runtime/recipe/particle.ts

```diff
@@ -339,7 +339,7 @@ export class Particle {
     return new Map();
   }
 
-  toString(nameMap, options) {
+  toString(nameMap, options): string {
```

Comment:
Should the second arg have the type `ToStringOptions`?

Reply:
ditto